owncloudApp/src/main/java/com/owncloud/android/shares/domain/OCShareRepository.kt

```diff
@@ -108,16 +100,92 @@ class OCShareRepository(
         password: String,
         expirationTimeInMillis: Long,
         publicUpload: Boolean
+    ): LiveData<Resource<Unit>> {
+        return insertShare(
+            filePath = filePath,
+            shareType = ShareType.PUBLIC_LINK,
+            permissions = permissions,
+            name = name,
+            password = password,
+            expirationTimeInMillis = expirationTimeInMillis,
+            publicUpload = publicUpload
+        )
+    }
+
+    override fun updatePublicShare(
+        remoteId: Long,
+        name: String,
+        password: String?,
+        expirationDateInMillis: Long,
+        permissions: Int,
+        publicUpload: Boolean
+    ): LiveData<Resource<Unit>> {
+        return updateShare(
+            remoteId,
+            permissions,
+            name,
+            password,
+            expirationDateInMillis,
+            publicUpload
+        )
+    }
+
+    /******************************************************************************************************
+     *********************************************** COMMON ***********************************************
+     ******************************************************************************************************/
+
+    private fun getShares(
+        filePath: String,
+        shareTypes: List<ShareType>
+    ): MutableLiveData<Resource<List<OCShare>>> {
+        return object : NetworkBoundResource<List<OCShare>, ShareParserResult>(appExecutors) {
+            override fun saveCallResult(item: ShareParserResult) {
+                val sharesFromServer = item.shares.map { remoteShare ->
+                    OCShare.fromRemoteShare(remoteShare).also { it.accountOwner = accountName }
+                }
+
+                if (sharesFromServer.isEmpty()) {
+                    localShareDataSource.deleteSharesForFile(filePath, accountName)
```

Comment:
Is this needed? Shouldn't be done inside 'replaceShares(...)' right below?

Reply:
The `replaceShares` method in `OCShareDao.kt` includes the code below and its aim is replacing the still existing shares, not deleting the shares that no longer exist in the server.

``` kotlin
@Transaction
open fun replaceShares(ocShares: List<OCShare>): List<Long> {
    for (ocShare in ocShares) {
        deleteSharesForFile(ocShare.path, ocShare.accountOwner)
    }
    return insert(ocShares)
}
```